src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/LocalizationModelDeserializer.cs

```
@@ -0,0 +1,230 @@
+        /// </list>
+        /// </summary>
+        private static IReadOnlyDictionary<string, ParameterChoiceLocalizationModel> LoadChoiceModels(IEnumerable<(IEnumerable<string> nameParts, string localizedString)> strings)
+        {
+            var results = new Dictionary<string, ParameterChoiceLocalizationModel>();
+
+            foreach (var choiceParts in strings.GroupBy(p => p.nameParts.FirstOrDefault()))
+            {
+                if (string.IsNullOrEmpty(choiceParts.Key))
+                {
+                    // Choice with no name. Ignore
+                    continue;
+                }
+
+                string? displayName = choiceParts.FirstOrDefault(p => p.nameParts.Skip(1).SingleOrDefault() == "displayName").localizedString;
+                string? description = choiceParts.FirstOrDefault(p => p.nameParts.Skip(1).SingleOrDefault() == "description").localizedString;
+
+                results.Add(choiceParts.Key, new ParameterChoiceLocalizationModel(displayName, description));
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Generates post action localization models from name parts such as:
+        /// <list type="table">
+        /// <item>postActions[0].description</item>
+        /// <item>postActions[0].manualInstructions[0].text</item>
+        /// <item>postActions[0].manualInstructions[1].text</item>
+        /// </list>
+        /// </summary>
+        private static IReadOnlyList<IPostActionLocalizationModel?> LoadPostActionModels(IEnumerable<(IEnumerable<string> nameParts, string localizedString)> strings)
+        {
+            var results = new List<IPostActionLocalizationModel?>();
+
+            foreach (var postActionParts in strings.GroupBy(p => p.nameParts.FirstOrDefault()))
+            {
+                if (!GetIndexFromString(postActionParts.Key, PostActionIndexPrefix, PostActionIndexSuffix, out int postActionIndex) ||
+                    postActionIndex > 256)
```

What does 256 represent, I understand its crazy high number, but I don't see much benefit having this here...

---

If someone accidentally types a high number in their localization file, I didn't want to allocate the memory for it. Can you think of any scenarios where this would hurt user experience?

---

I'm not following, where is memory allocated for it? I'm just not a fan of complicating code with some artificial limit and in theory limiting template post actions limit, I agree nobody should reach this limit but still...

---

Right below this if statement:
```csharp
while (results.Count <= postActionIndex)
{
    // We haven't processed localizations for these post actions. Just put null for now.
    // Localization file also may not specify translations for all the post actions.
    results.Add(null);
}
```

---

This won't limit the number of post actions, but anything after 256 wouldn't be localized.

---

As we discussed internally, this is not swallowing an error. This is a valid use case. I changed the implementation to use dictionary.